R/calibrate.R

```diff
@@ -606,6 +553,99 @@ calibrate_outputs <- function(output,
   output
 }
 
+#' Helper functions for implementing calibration
+#'
+#' @param proportion_raw vector of initial input proportions
+#' @param denominator_new vector of denomin
+#' @param target_val single value for the adjusted total
+#'
+#' @description
+#' These functions calibrate input proportions so that
+#' `sum(proportion_adj * denominator_new) == target_val`,
+#' where `proportion_adj` is adjusted results of input
+#' `proportion_raw` adjusted by a fixed odds ratio.
+#'
+#' For function interface convenience, the parameter `target_val`
+#' may be a single number or a vector with the same value
+#' repeated (typically of same length as `proportion_raw` and
+#' `denominator_new`), but the value must be identical and only
+#' the first value of the vector will be used.
+#'
+#' `calculate_logit_adjustment()` returns the calculated logit
+#' adjustment value. 
+#' 
+#' `calibrate_logistic_one()` returns the adjusted vector (same
+#' length as `proportion_raw`.
+#'
+#' @noRd
+calculate_logit_adjustment <- function(proportion_raw,
+                                       denominator_new,
+                                       target_val) {
+    
+  stopifnot(proportion_raw >= 0)
+  stopifnot(proportion_raw <= 1)
+  stopifnot(target_val == target_val[1])
```

Comment:
How likely are these to error? I'd prefer in general to use an `if (..) { stop("...") }` instead of `stopifnot` because then we can raise a better error message. We should raise translated error messages using `stop(t_("ERROR_KEY"))` if this is going to be surfaced to users.

Reply:
I don't expect these errors to occur in normal use.So not so concerned about about something interpretable to user.


If they do, it indicates something pathological has happened elsewhere in the Spectrum file, or a case that we haven't thought of. 
 
 (This reminds me though; I want to test this on a large batch of Spectrum files before we merge and deploy, and also ensure Rachel has reviewed. Will edit the PR message to note that.)